types/marv/api/callback.d.ts

```diff
@@ -0,0 +1,28 @@
+/// <reference types="node" />
+
+import { PathLike } from 'fs';
+import { Driver, EmptyObject, ErrorOnlyCallback, ParsedMigration, ScanOptions } from '../index';
```

Comment:
make 'index.d.ts' contain only types (not symbols like const/class), and use:
```ts
import type {...} ...
```
in your submoudles (api/callback, api/promise). The `import type ` syntax is now supported by default on DT repo (TS 3.9 is the lowest version in support version window)